### PR TITLE
Did exchange test dev acapy

### PR DESCRIPTION
**Pull request by nodlesh**
Working initial Base test scenario for DID Exchange for acapy.  There are still some outstanding questions about inconsistencies between The RFC, Acapy implementation, and the Connection protocol, but I'll sort those when I next talk to sklump. 

@murpheux This submission should be close to final Test to Backchannel API, for you to finalize against for afgo.

### CODE REVIEW

File: aries-backchannels/acapy/acapy_backchannel.py [hunk 1]
```diff
@@ -100,6 +100,26 @@ def __init__(
             "presentation_acked": "done"
         }
 
+        # Aca-py : RFC
+        self.didExchangeResponderStateTranslationDict = {
+            "initial": "invitation-sent", # responder
+            "invitation": "invitation-received", # requester
```
Comment (swcurran) [inline aries-backchannels/acapy/acapy_backchannel.py]: Curious -- why does this comment say "requester" -- these are all responder states/actions

File: aries-backchannels/acapy/acapy_backchannel.py [hunk 2]
```diff
@@ -100,6 +100,26 @@ def __init__(
...
+            "?": "abandoned",
+            "active": "completed"
+        }
+
+        # Aca-py : RFC
+        self.didExchangeRequesterStateTranslationDict = {
+            "initial": "invitation-sent", # responder
+            "invitation": "invitation-received", # requester
+            "request": "request-sent", # requester
+            "response": "response-received", # requester
```
Comment (swcurran) [inline aries-backchannels/acapy/acapy_backchannel.py]: Same as above.  In this case, the requester doesn't know when this happens.

File: aries-backchannels/backchannel_operations.csv [hunk 3]
```diff
@@ -87,11 +87,16 @@ rfc,command,response,topic,method,operation,id,data,description,id_details,data_
 0011 Revocation,X,,revocation-registry,GET,,Y,,Get revocation info from issuers webhook ,thread_id,,revocation_registry
 0036 Issue Credential,X,,credential,GET,,Y,,Get a stored credential from the holders wallet ,credential_id,,credential
 0036 Issue Credential,X,,credential,GET,revoked,Y,,Get a credentials revoked status ,credential_id,,revoked
+0434 Out of Band,X,,out-of-band,POST,send-invitation-message,,Y,Create an OOB invitation ,,invitation,invitation
+0434 Out of Band,X,,out-of-band,POST,receive-invitation,,Y,Receive invitation,,invitation,"state, connection_id"
```
Comment (swcurran) [inline aries-backchannels/backchannel_operations.csv:91]: We'll need to add an "out of band" message for "reuse" eventually.  Not yet in OOB yet, but coming soon.

File: aries-test-harness/features/0023-did-exchange.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Feature: Establishing DID Based Relationships RFC 0023
+
+   @T001-RFC0023 @P1 @critical @AcceptanceTest @RFC0023 @wip
+   Scenario: Establish a DID based connection between two agents with an explicit invitation
```
Comment (swcurran) [inline aries-test-harness/features/0023-did-exchange.feature]: The title of this is not great.  The connection is "DID based" regardless of whether 0160 of 0023 DID Exchange is used.  I'd recommend being more explicit about saying this is using 0023 DID Exchange (assuming it is).

**Review by swcurran — COMMENTED** [review]
Some minor issues.  Ran the test and looks good.

File: aries-backchannels/acapy/acapy_backchannel.py (hunk 1 above)
Comment (nodlesh) [reply aries-backchannels/acapy/acapy_backchannel.py]: Copy/Paste error. Fixed.

File: aries-backchannels/acapy/acapy_backchannel.py (hunk 2 above)
Comment (nodlesh) [reply aries-backchannels/acapy/acapy_backchannel.py]: Copy/Paste error.

File: aries-test-harness/features/0023-did-exchange.feature (hunk 4 above)
Comment (nodlesh) [reply aries-test-harness/features/0023-did-exchange.feature]: Title and other "DID based" references changed to DID Exchange explicitly. Also modified the final check for a successful connection to be a common step between both Connection protocol and DID Exchange. Unfortunately the states in the RFC for these two are different so the existing check of "complete" could not be used.

**Review by swcurran — COMMENTED** [review]
LGTM - tests still run.

**Review by swcurran — APPROVED** [review]

**Pull request merged**